notification-bot/src/index.ts

```diff
@@ -0,0 +1,215 @@
+  return Filter.create()
+    .withHeader({ weak: true })
+    .addEvent(new EventFilter().withFromAddress(governance_contract_address))
+    .encode();
+}
+
+/**
+ * Configures the Apibara client with a specific filter and block number.
+ * @param {StreamClient} client The Apibara client instance to configure.
+ * @param {string} filter The encoded filter specifying the events to listen for.
+ * @param {number} block_number The blockchain block number from which to start listening.
+ */
+function configureClient(client: StreamClient, filter: Uint8Array, block_number: number) {
+  client.configure({
+    filter: filter,
+    batchSize: 1,
+    cursor: StarkNetCursor.createWithBlockNumber(block_number),
+  });
+}
+
+/**
+ * Main entry point for the event monitoring script.
+ * Initializes the StreamClient and RpcProvider for connecting to the StarkNet blockchain.
+ * Configures the Apibara client to listen for specific blockchain events and processes them accordingly.
+ */
+async function main() {
+  try {
+    // Apibara streaming
+    const client = initializeStreamClient();
+
+    const provider = new RpcProvider({
+      nodeUrl: constants.NetworkName.SN_SEPOLIA,
+      chainId: constants.StarknetChainId.SN_SEPOLIA,
+    });
+
+    const hashAndBlockNumber = await provider.getBlockLatestAccepted();
+    const block_number = hashAndBlockNumber.block_number;
+
+    //Initialize the filter
+    const filter_test = initializeFilter();
+
+    // Configure the apibara client
+    configureClient(client, filter_test, block_number);
+    await listenToMessages(client);
+
+  } catch (error) {
+    console.error("Initialization failed", error);
+    process.exit(1);
+  }
+}
+
+/**
+ * Listens to blockchain events and handles each submit proposel event as they are received.
+ * @param {StreamClient} client The Apibara client instance from which to listen for messages.
+ */
+async function listenToMessages(client: StreamClient) {
+  for await (const message of client) {
+    if (message.message === "data" && message.data?.data) {
+      for (const data of message.data.data) {
+        const block = v1alpha2.Block.decode(data);
+
+        const { header, events, transactions } = block;
+
+        if (!header || !transactions) {
+          continue;
+        }
+
+        for (const event of events) {
+          console.log(event.event)
+          if (event.event && event.event.keys && event.event.data) {
+            for (let evtkey of event.event!.keys) {
+              let evtkey_hex = FieldElement.toHex(evtkey);
+
+              let submit_proposal_hash = PROPOSAL_HASH;
+              let isSame = evtkey_hex === submit_proposal_hash;
+
+              if (isSame) {
+                handleEventSubmitProposal(header, event.event);
+              }
+            }
+          }
+          break;
+        }
+      }
+    }
+  }
+}
+
+
+main()
+  .then(() => process.exit(0))
+  .catch((error) => {
+    console.error(error);
```

Comment:
The bot should attempt to send the error to Telegram as nobody will be watching the output. We use this for monitoring elsewhere too.

Reply:
done